packages/client/tests/functional/_utils/getTestSuitePlan.ts

```diff
@@ -65,6 +65,11 @@ function getExpandedTestSuitePlanWithProviderFlavors(suiteConfig: NamedTestSuite
     newSuiteConfig.parametersString += `, ${flavor}`
     // ^^^ temporary until I get to the TODO in getTestSuiteParametersString
 
+    // if the test is not doing stuff with relation mode already, we set one
+    if (newSuiteConfig.matrixOptions.relationMode === undefined) {
+      newSuiteConfig.matrixOptions.relationMode = relationModesForFlavor[flavor]
+    }
+
```

Comment:
This ensures that a given flavor will have what it needs to work, or be skipped when needed.